### PR TITLE
Prep for 0.19: remove `toString` calls

### PR DESCRIPTION
**Pull request by stoeffel**
Looks like tests on our master are broken.

### CODE REVIEW

File: examples/e2e-tests/elm/Generated/Api.elm [hunk 1]
```diff
@@ -133,42 +148,46 @@ postPost body =
         , withCredentials =
             False
         }
+        <<<<<<< HEAD
```
Comment (joneshf) [inline examples/e2e-tests/elm/Generated/Api.elm]: Bad merge?

File: src/Servant/Elm/Internal/Generate.hs [hunk 2]
```diff
@@ -450,15 +465,63 @@ isEmptyType :: ElmOptions -> ElmDatatype -> Bool
...
 -}
 isElmStringType :: ElmOptions -> ElmDatatype -> Bool
+isElmStringType _ (ElmPrimitive (EList (ElmPrimitive EChar))) = True
 isElmStringType opts elmTypeExpr =
   elmTypeExpr `elem` stringElmTypes opts
 
 
+toStringSrcTypes :: T.Text -> ElmOptions -> ElmDatatype -> T.Text
+toStringSrcTypes operator opts (ElmPrimitive (EMaybe argType)) = toStringSrcTypes operator opts argType
+toStringSrcTypes _ _ (ElmPrimitive (EList (ElmPrimitive EChar))) = "identity"
```
Comment (joneshf) [inline src/Servant/Elm/Internal/Generate.hs]: It seems like this case would give a different type from the other cases, but I don't really understand how this function is supposed to work 😅. What does it do?

File: src/Servant/Elm/Internal/Generate.hs [hunk 3]
```diff
@@ -450,15 +465,63 @@ isEmptyType :: ElmOptions -> ElmDatatype -> Bool
...
+  -- Don't use "toString" on Elm Strings, otherwise we get extraneous quotes.
+  | isElmStringType opts argType = stext ""
+  | otherwise                    = stext $ toStringSrcTypes operator opts argType <> " " <> operator
+
 
 {- | Determines whether we call `toString` on URL captures and query params of
 this type in Elm.
 -}
 isElmStringType :: ElmOptions -> ElmDatatype -> Bool
+isElmStringType _ (ElmPrimitive (EList (ElmPrimitive EChar))) = True
```
Comment (joneshf) [inline src/Servant/Elm/Internal/Generate.hs]: Is this saying that an elm `List Char` is an elm `String`? If so, how does that work?

File: src/Servant/Elm/Internal/Generate.hs [hunk 4]
```diff
@@ -40,9 +40,16 @@ data ElmOptions = ElmOptions
     -- ^ Types that represent an empty Http response.
   , stringElmTypes        :: [ElmDatatype]
     -- ^ Types that represent a String.
+  , intElmTypes        :: [ElmDatatype]
```
Comment (joneshf) [inline src/Servant/Elm/Internal/Generate.hs:43]: Neat!

**Review by joneshf — COMMENTED** [review]
Wooo!!!

Had some questions about the char stuff, and I think a typo from a bad merge. Otherwise looks good.

If it's all good, feel free to merge without future review after the merge fix!

File: examples/e2e-tests/elm/Generated/Api.elm (hunk 1 above)
Comment (stoeffel) [reply examples/e2e-tests/elm/Generated/Api.elm]: oops

File: src/Servant/Elm/Internal/Generate.hs (hunk 3 above)
Comment (stoeffel) [reply src/Servant/Elm/Internal/Generate.hs]: yes. don't know how it works, but it does 🙂

File: src/Servant/Elm/Internal/Generate.hs (hunk 2 above)
Comment (stoeffel) [reply src/Servant/Elm/Internal/Generate.hs]: well document.

Comment (stoeffel) [reply src/Servant/Elm/Internal/Generate.hs]: *will

**Comment by stoeffel** [comment]
waiting with merge until I get the integration tests green.

**Comment by stoeffel** [comment]
tests are broken on master, because of the last two commits. I've fixed all the tests in here, but I had to add the orphan instance back. I didn't want to deal with that in this PR. happy to pair on removing that again in the future.
@joneshf would love another review.

File: src/Servant/Elm/Internal/Orphans.hs [hunk 5]
```diff
@@ -12,3 +12,8 @@ instance ElmType ElmDatatype where
 
 
 instance ElmType NoContent
+
+-- TODO: Generate Elm functions that can handle the response headers. PRs
+-- welcome!
+instance (ElmType a) => ElmType (Headers ls a) where
```
Comment (joneshf) [inline src/Servant/Elm/Internal/Orphans.hs]: I don't think we want to add this back.

We moved the instance to `elm-export`: https://github.com/NoRedInk/elm-export/commit/4b68744a3af70f75ed7d5cb6782454d87cdcd3d3.

If we add this back, we've no clue which instance ghc will chose when it decides to choose one. We can hope and pray that the two instances line up, but if one of them changes, we're really messed up when ghc decides which instance to choose.

I think we should drop this instance again and do one of the following:
* Depend on our fork of `elm-export`.
    It doesn't look like our changes are going to be merged into upstream anytime soon, so no sense trying to keep parity with it. The tests will start compiling then.
* Remove the test for `Headers`.
    If we can't support this data type, we shouldn't test against it.

No preference for either option, but I don't think re-adding this orphan is the way to go.

**Review by joneshf — CHANGES_REQUESTED** [review]
Sorry about breaking the tests.

Aside from the orphan stuff, everything looks good. Left some comments on it locally.

File: src/Servant/Elm/Internal/Orphans.hs (hunk 5 above)
Comment (joneshf) [reply src/Servant/Elm/Internal/Orphans.hs]: Aside, this is a prime example of why working with these two dependencies has been frustrating for us. You start out wanting to make a seemingly innocuous change, and it ends up expanding into a whole web of problems.

Comment (stoeffel) [reply src/Servant/Elm/Internal/Orphans.hs]: thanks for the explanation and proposed solutions. will address asap.

Comment (stoeffel) [reply src/Servant/Elm/Internal/Orphans.hs]: @joneshf I removed the orphan and change the dependency to use our fork.

**Review by joneshf — APPROVED** [review]

**Comment by joneshf** [comment]
Looks great!

**Pull request merged**